### ISSUE TITLE
Add support for more dimensions -  ImageDataGenerator's .flow_from_directory()

### ISSUE THREAD

**Issue by maxvfischer** [tensorflow/tensorflow#39402]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.1.0
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
When using ImageDataGenerator's .flow_from_directory(), one has to specify color_mode ∈ {"grayscale", "rgb", "rgba"} (1, 3 or 4 dim).

I propose to change the color mode to an input_shape() to be able to handle more dimensions.

**Will this change the current api? How?**
Either remove the color_mode completely and handle it dynamically or let the user input the input_shape of the data.

**Who will benefit with this feature?**
When working with multiclass image segmentations and an one-hot encoded representations of the segmentation masks, it is very convenient to use the ImageDataGenerator's .flow_from_directory() for both the image and the mask, then zip them when inputted to model.fit(). 

Today, this does only work if the mask is of dim ∈ {1, 3, 4}.

One way to circumvent this is to write your own generator by inheriting from keras.utils.Sequence class, but it would be nice to have support for it in ImageDataGenerator as one then can use the built-in augmentations.

**Any Other info.**
Related question on Stackoverflow: https://stackoverflow.com/questions/60551136/how-to-use-imagedatagenerator-with-multi-label-masks-for-multi-class-image-segme

**Comment by jvishnuvardhan** [comment]
@maxvfischer Sorry for the late response. Are you still interested in contributing? If yes, please feel free to open a PR in  [keras-team/keras repo.](https://github.com/keras-team/keras/issues) repository.

Please note that Keras development moved to keras-team/keras repository to focus on only keras. Thanks!
